session.go

```diff
@@ -416,12 +429,13 @@ func (s *Session) mux(remoteBegin *frames.PerformBegin) {
 						}
 					}
 
-					link, ok := links[handle]
```

Comment:
`links` was mapping input (i.e. remote) handles to `*link`, but since we populated `handlesByDeliveryID` with the _output_ (i.e. our) handle, we won't find the entry if the input and output handles don't match.